File: FitTrack-Frontend/src/app/responses/user/get-user-details.response.ts
import { Guid } from 'guid-typescript';
import { GenderEnum } from '../../enums/gender.enum';
import { RoleEnum } from '../../enums/role.enum';

export interface GetUserDetailsResponse {
  id: Guid;
  username: string;
  profilePictureUrl: string;
  registrationDate: Date;
  isEmailConfirmed: boolean;
  firstName: string;
  lastName: string;
  email: string;
  role: RoleEnum;
  gender: GenderEnum;
  height: number;
  weight: number;
}

File: FitTrack-Backend-old/FitTrack.Data.Contract/Helpers/Responses/GetUserDetailsResponse.cs
using FitTrack.Data.Object.Enums;

namespace FitTrack.Data.Contract.Helpers.Responses;

public class GetUserDetailsResponse
{
    public Guid Id { get; set; }
    public required string Username { get; set; }
    public string? ProfilePictureUrl { get; set; }
    public DateTime RegistrationDate { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string Email { get; set; }
    public bool IsEmailConfirmed { get; set; }
    public RoleEnum Role { get; set; }
    public GenderEnum Gender { get; set; }
    public double Height { get; set; }
    public double Weight { get; set; }
}
